src/hubspot/profile.ts

```diff
@@ -0,0 +1,156 @@
+   *
+   * @default Retrieves a fresh accessToken with the information in the [flow property]{@link HubSpotOAuthSettings#flow}
+   */
+  readonly accessToken?: SecretValue;
+
+  /**
+   * The OAuth flow used for obtaining a new accessToken when the old is not present or expired.
+   *
+   * @default undefined. AppFlow will not request any new accessToken after expiry.
+   */
+  readonly flow?: HubSpotOAuthFlow;
+
+  /**
+   * The OAuth token and authorization endpoints.
+   */
+  readonly endpoints?: HubSpotOAuthEndpoints;
+}
+
+export class HubSpotConnectorProfile extends ConnectorProfileBase {
+  public static fromConnectionProfileArn(
+    scope: Construct,
+    id: string,
+    arn: string,
+  ) {
+    return this._fromConnectorProfileAttributes(scope, id, {
+      arn,
+    }) as HubSpotConnectorProfile;
+  }
+
+  public static fromConnectionProfileName(
+    scope: Construct,
+    id: string,
+    name: string,
+  ) {
+    return this._fromConnectorProfileAttributes(scope, id, {
+      name,
+    }) as HubSpotConnectorProfile;
+  }
+
+  private static readonly defaultTokenEndpoint: string =
+    "https://api.hubapi.com/oauth/v1/token";
+
+  constructor(
+    scope: Construct,
+    id: string,
+    props: HubSpotConnectorProfileProps,
+  ) {
+    super(scope, id, props, HubSpotConnectorType.instance);
+  }
+
+  protected buildConnectorProfileProperties(
+    props: ConnectorProfileProps,
+  ): CfnConnectorProfile.ConnectorProfilePropertiesProperty {
+    const properties = props as HubSpotConnectorProfileProps;
+    return {
+      customConnector: {
+        oAuth2Properties: {
+          // INFO: even if we're using a refresh token grant flow this property is required
+          oAuth2GrantType: OAuthGrantType.AUTHORIZATION_CODE,
+          // INFO: even if we provide only the access token this property is required
+          // TODO: think about if this is correct. token can be IResolvable
+          tokenUrl:
+            properties.oAuth.endpoints?.token ??
+            HubSpotConnectorProfile.defaultTokenEndpoint,
+        },
+      },
+    };
+  }
+
+  protected buildConnectorProfileCredentials(
+    props: ConnectorProfileProps,
+  ): CfnConnectorProfile.ConnectorProfileCredentialsProperty {
+    const properties = props as HubSpotConnectorProfileProps;
+
+    return {
+      customConnector: {
+        oauth2: {
+          // INFO: when using Refresh Token Grant Flow - access token property is required
+          accessToken:
+            properties.oAuth.accessToken?.unsafeUnwrap() ?? "dummyAccessToken",
+          refreshToken:
+            properties.oAuth.flow?.refreshTokenGrant.refreshToken?.unsafeUnwrap(),
+          clientId:
+            properties.oAuth.flow?.refreshTokenGrant.clientId?.unsafeUnwrap(),
+          clientSecret:
+            properties.oAuth.flow?.refreshTokenGrant.clientSecret?.unsafeUnwrap(),
```

Comment:
per CDK convention, mark uses of `unsafeUnwrap` where safe with a `// Safe usage` comment for clarity.

Reply:
created separate issue as it affects multiple connectors